src/components/ArenaProofModal.js

```diff
@@ -20,131 +19,125 @@ import { uploadToS3 } from "../libs/s3";
-												this.props.day,
-												this.state.key,
-												this.props.activeSprintId,
-												`${this.props.user.fName} just completed ${
-													this.props.activeMission.title
-												}.${
-													this.state.trashTalk.length > 0
-														? `They also said: "${this.state.trashTalk}"`
-														: ""
-												} `
-											);
-											this.setState({ key: "", trashTalk: "" });
-											this.props.handleClose();
-										}}
-										bsSize="large"
-										text={I18n.get("submitProof")}
-										loadingText={I18n.get("creating")}
-										// disabled={!this.validateForm()}
-										isLoading={this.props.loading}
-									/>
-								</div>
-							</React.Fragment>
-						) : (
-							<div>
-								{this.props.activeMission.proofLink !== "" ? (
-									<a
-										href={`https://${process.env.REACT_APP_PROOF_BUCKET}-prod.s3.amazonaws.com/public/${this.props.activeMission.proofLink}`}
-										target="_blank"
-										rel="noopener noreferrer"
-									>
-										{I18n.get("viewProof")}
-									</a>
-								) : (
-									<p>{I18n.get("noProof")}</p>
-								)}
+	return (
+		<div>
+			<Modal show={show} onHide={handleClose}>
+				<Modal.Header closeButton>
+					<Modal.Title>{I18n.get("submitProof")}</Modal.Title>
+				</Modal.Header>
+				<Modal.Body>
+					{view === "submit" ? (
+						<React.Fragment>
+							<FormGroup bsSize="large" controlId="trashTalk">
+								<ControlLabel>{I18n.get("trashTalkPSA")}</ControlLabel>
+								<FormControl
+									type="text"
+									onChange={handleChange}
+									value={formData.trashTalk}
+								/>
+							</FormGroup>
+							<h3>{I18n.get("attachment")}</h3>
+							<input type="file" onChange={(evt) => onChange(evt)} />
+							<br />
+							<div className="flex">
+								<Button onClick={handleClose}>{I18n.get("close")}</Button>
+								<LoaderButton
+									onClick={() => {
+										propsHandleChange(
+											activeMission,
+											activeIndex,
+											day,
+											key,
+											activeSprintId,
+											`${user.fName} just completed ${activeMission.title}.${
+												formData.trashTalk.length > 0
+													? `They also said: "${formData.trashTalk}"`
+													: ""
+											} `
+										);
+										setFormData({ trashTalk: "" });
+										setKey("");
+										handleClose();
+									}}
+									bsSize="large"
+									text={I18n.get("submitProof")}
+									loadingText={I18n.get("creating")}
+									// ! Is there a reason this is commented?
+									// disabled={!this.validateForm()}
+									isLoading={loading}
+								/>
 							</div>
-						)}
-					</Modal.Body>
-					<Modal.Footer />
-				</Modal>
-			</div>
-		);
-	}
+						</React.Fragment>
+					) : (
+						<div>
+							{activeMission.proofLink !== "" ? (
+								<a
+									href={`https://${process.env.REACT_APP_PROOF_BUCKET}-prod.s3.amazonaws.com/public/${activeMission.proofLink}`}
```

Comment:
You do not need the extra `-prod` - this has actually been fixed in recent code that I personally pushed up. You need to run `git pull` from my upstream main repository, and pull that code into your code base. You will likely have some merge conflicts. Then, I need you to manually test all of this to see that it works. Right now, looking at this code, I know for certain it would not work if I tested it. 

This comes back to the mantra - never send broken code in a PR. At the VERY LEAST, label it as a 'Draft PR', and explicitly ask questions about 'hey what about this' - that is perfectly acceptable and encouraged.

Reply:
Ok I've tested this, and the form state works and I've also tried submission(works successfully without any errors.

Reply:
You have not actually fixed this - the `-prod` is still there when I pull down the code.

Reply:
If you scroll down and check to see the 'proof' of the modal on the anchor tag, you will see that it takes you to undefined.

Reply:
Oh ok, I did not understand that you were taking about the `-prod` in the link

Reply:
Hey @mikhael28,

I did not initially understand what you said about `-prod`, I've resolved that now. Also, about the key being undefined:

>After some more digging, I have gone ahead and disabled the s3 upload for the new submit proof modal. It turns out that we were not saving the file type, as we were in the Arena proof modal - and that, while things did work in the past (somehow), they are not working now. None of that is your fault, all legacy decisions made a while ago. I've gone ahead and disabled it, and am going to re-enable it later. Considering pushing the curriculum data structure revamp much closer in my timeline.

Shouldn't we do the same thing here if we face the same issue here?